crates/bevy_app/src/app.rs

```diff
@@ -115,6 +139,7 @@ impl App {
         for sub_app in self.sub_apps.values_mut() {
             (sub_app.runner)(&mut self.world, &mut sub_app.app);
         }
+        self.has_run = true;
```

Comment:
I think it's better to put this in `App::run` instead of setting the value for every update.

Reply:
Yep; I use App::update quite commonly for tests and would also appreciate a warning there.